pkg/store/prometheus.go

```diff
@@ -336,10 +336,39 @@ func extendLset(lset []storepb.Label, extend labels.Labels) []storepb.Label {
 }
 
 // LabelNames returns all known label names.
-func (p *PrometheusStore) LabelNames(ctx context.Context, r *storepb.LabelNamesRequest) (
+func (p *PrometheusStore) LabelNames(ctx context.Context, _ *storepb.LabelNamesRequest) (
 	*storepb.LabelNamesResponse, error,
 ) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	u := *p.base
+	u.Path = path.Join(u.Path, "/api/v1/labels")
+
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		return nil, status.Error(codes.Unknown, err.Error())
```

Comment:
When Context canceled error case occurs we should check if partial response enabled and return?

Also instead of `codes.Unknown` we should map HTTP error codes to grpc. i.e. if prometheus returns 500x we should return StatusUnavailable.

Reply:
> Also instead of codes.Unknown we should map HTTP error codes to grpc. i.e. if prometheus returns 500x we should return StatusUnavailable.

The error returned by `http.NewRequest()` is only occured with url parsing error or invalid http method, we could not pay attention to this error. 

Agress with mapping code by the response of real http request with Prometheus.

Reply:
> When Context canceled error case occurs we should check if partial response enabled and return?

Sorry, I could not get your point, IMO, if the http request is cancelled or timeout, there is no data returned, there is only one instance of Prometheus would be requested.

Reply:
You are right in this place, it's a parsing error, so no context at play here.